### ISSUE TITLE
InitPointCloudPacked test failed on i386

### ISSUE THREAD

**Issue by osrf-migration** [gazebosim/gz-msgs#32]
**Original report ([archived issue](https://osrf-migration.github.io/ignition-gh-pages/#!/ignitionrobotics/ign-msgs/issues/32)) by Jose Luis Rivero (Bitbucket: [Jose Luis Rivero](https://bitbucket.org/%7Bd12309b2-f745-42ee-b119-aec4fcdf81fe%7D/), GitHub: [j-rivero](https://github.com/j-rivero)).**

----------------------------------------
# Prerequisites

‌    \[x\] Put an X between the brackets on this line if you have done all of the following:

‌        \* Checked the Q&amp;A board for common solutions: http://answers.gazebosim.org

‌        \* Checked that your issue isn&#39;t already filed.

‌        \* Checked that there is not already an Ignition package that provides the described functionality: https://ignitionrobotics.org/libs

# Description

The InitPointCloudPacked is failing on i386. I&#39;ve detected it in the i386 package build: [https://build.osrfoundation.org/view/main/view/ignition/job/ign-msgs4-debbuilder/167/console](https://build.osrfoundation.org/view/main/view/ignition/job/ign-msgs4-debbuilder/167/console)

```
[ RUN      ] UtilityTest.InitPointCloudPacked
/var/lib/jenkins/workspace/ign-msgs4-debbuilder/build/ignition-msgs-4.5.0/src/Utility_TEST.cc:651: Failure
Expected equality of these values:
  24u
    Which is: 24
  pc.point_step()
    Which is: 16
/var/lib/jenkins/workspace/ign-msgs4-debbuilder/build/ignition-msgs-4.5.0/src/Utility_TEST.cc:669: Failure
Expected equality of these values:
  16u
    Which is: 16
  pc.field(3).offset()
    Which is: 12
[  FAILED  ] UtilityTest.InitPointCloudPacked (0 ms)
```

# Steps to Reproduce

1. Build the software on i386 and run the test suite

**Expected behavior:**

Pass the test

**Actual behavior:**

Failing test

**Reproduces how often:**

always

# Versions

ubuntu bionic i386 for 4.5.0 msgs version

**Comment by richmattes** [comment]
I'm seeing this on Fedora as well.  Without digging in too deeply, it looks like the struct padding for packed point clouds is based on `sizeof(size_t)`, which is 32 on 32-bit platforms and 64 on 64-bit platforms.

**Comment by scpeters** [comment]
I believe this has been fixed by https://github.com/ignitionrobotics/ign-msgs/pull/120

**Comment by scpeters** [comment]
> I believe this has been fixed by #120

this was included in 6.3.0, though it's not in the changelog
